### ISSUE TITLE
FAQs: Images of the Threat Diagram are cropped in the Report

### ISSUE THREAD

**Issue by craig-Ayla** [mike-goodwin/owasp-threat-dragon-desktop#146]
When I go to run the Report on a model (actually a group of models), the Diagrams are cropped so that the reader can't see the entire model.

I'm getting several red, brief, pop-up style warnings when running the Report.  They appear to be identical and state "Error: cannot read property 'cells' of undefined".

I'm using version 1.2, on a MacBook Air running 10.14.6

**Comment by jgadsden** [comment]
Hello @craig-Ayla - could you send me either the pdf output or the json file? I can have a look
Having said that I understand if you do not want to send your threat models around for everyone to have a look ... a bit of a reveal :-) But if you can repeat it with a generic model that would be great

**Comment by jgadsden** [comment]
I see the error "Error: cannot read property 'cells' of undefined" when I have a diagram with no elements in it, do you have any diagrams in your threat model that are empty?

**Comment by craig-Ayla** [comment]
No.  The diagrams all have elements (process, actor, store…).  There are 6 diagrams.

Some of the diagrams do not have Threats yet.

Craig




On Jul 13, 2020, at 10:09 PM, Jon Gadsden <notifications@github.com<mailto:notifications@github.com>> wrote:



I see the error "Error: cannot read property 'cells' of undefined" when I have a diagram with no elements in it, do you have any disgrams in your threat model that are empty?

**Comment by craig-Ayla** [comment]
Let me see if I can recast it so it’s more generic…  Probably be a day or two (sorry - can’t be helped).

Craig




On Jul 13, 2020, at 10:08 PM, Jon Gadsden <notifications@github.com<mailto:notifications@github.com>> wrote:



Hello @craig-Ayla<https://github.com/craig-Ayla> - could you send me either the pdf output or the json file? I can have a look
Having said that I understand if you do not want to send your threat models around for everyone to have a look ... a bit of a reveal :-) But if you can repeat it with a generic model that would be great

**Comment by jgadsden** [comment]
OK, thanks @craig-Ayla - it may be platform dependent as well, so I can test your json/pdf on all three platform types when you are ready.

Not sure why you are getting the errors "Error: cannot read property 'cells' of undefined". Do you want to raise a bug issue for this at : https://github.com/OWASP/threat-dragon-core/issues

Thanks again, Jon

**Comment by craig-Ayla** [comment]
Hi Jon,

I’m terribly sorry - there were diagrams without any objects in them.  I’d forgotten I’d added them.

Also, I didn’t realize that the ends of the Data Flow arrows could be ‘attached’ to the Actor/Store, etc by dragging them into the center of the object.  Fixing that had the side-effect of fixing the diagram cropping issue.

What would you like me to do with the Issue on GitHub now?

Craig




On Jul 14, 2020, at 9:32 AM, Jon Gadsden <notifications@github.com<mailto:notifications@github.com>> wrote:



OK, thanks @craig-Ayla<https://github.com/craig-Ayla> - it may be platform dependent as well, so I can test your json/pdf on all three platform types when you are ready.

Not sure why you are getting the errors "Error: cannot read property 'cells' of undefined". Do you want to raise a bug issue for this at : https://github.com/OWASP/threat-dragon-core/issues

Thanks again, Jon

**Comment by jgadsden** [comment]
No problem at all @craig-Ayla , good to hear that the cropping issue is sorted and that the cause of the errors are known. We need a FAQ on this really, because many users would have been in the same situation. Very good that you raised this issue, otherwise we never hear about these problems.

Do you mind if we keep this as a Documentation issue? We need to update the FAQs for this ... possibly on the project page at https://owasp.org/www-project-threat-dragon/

**Comment by craig-Ayla** [comment]
Feel free to keep everything I’ve sent.  And, thanks for providing the tool!

Craig




On Jul 14, 2020, at 12:41 PM, Jon Gadsden <notifications@github.com<mailto:notifications@github.com>> wrote:



No problem at all @craig-Ayla<https://github.com/craig-Ayla> , good to hear that the cropping issue is sorted and that the cause of the rrors are known. We need a FAQs really, because many users would have been in the same situation. Very good that you raised this issue, otherwise we never hear about these problems.

Do you mind if we keep this as a Documentation issue? We need to update the FAQs for this ... possibly on the project page at https://owasp.org/www-project-threat-dragon/

**Comment by craig-Ayla** [comment]
Hey, I do have one question (or possibly request).  I’m using the local version.   I open the model file, then select a diagram and work on it, and often I want to go to a different diagram.  It seems that I have to completely close the model, then reopen it, and then select the next diagram I want to work on.  I’m wondering if there’s a way that I can go from diagram to diagram without closing the model file and reopening it.  If there’s not, I think that would be a good usability enhancement.

Craig




On Jul 14, 2020, at 4:01 PM, Craig Payne <craig@aylanetworks.com<mailto:craig@aylanetworks.com>> wrote:

Feel free to keep everything I’ve sent.  And, thanks for providing the tool!

Craig




On Jul 14, 2020, at 12:41 PM, Jon Gadsden <notifications@github.com<mailto:notifications@github.com>> wrote:



No problem at all @craig-Ayla<https://github.com/craig-Ayla> , good to hear that the cropping issue is sorted and that the cause of the rrors are known. We need a FAQs really, because many users would have been in the same situation. Very good that you raised this issue, otherwise we never hear about these problems.

Do you mind if we keep this as a Documentation issue? We need to update the FAQs for this ... possibly on the project page at https://owasp.org/www-project-threat-dragon/

**Comment by jgadsden** [comment]
Hello @craig-Ayla - there is a quicker way to swap between diagrams. Exit from the first diagram using the 'X' button (between the grid and the bin buttons) in the drawing pane. This takes you back to the project view, and then click on the next diagram thumbnail.

Again this is good feedback, and I will put it into the FAQs.

Cheers, Jon

**Comment by craig-Ayla** [comment]
Ah ha!  That works like a charm, though it’s not what I would have expected for a button that says ‘cancel’ when I hover over it.  Maybe ‘cancel pending edits and close diagram’?

Craig




On Jul 15, 2020, at 12:28 PM, Jon Gadsden <notifications@github.com<mailto:notifications@github.com>> wrote:



Hello @craig-Ayla<https://github.com/craig-Ayla> - there is a quicker way to swap between diagrams. Exit from the first diagram using the 'X' button (between the grid and the bin buttons) in the drawing pane. This takes you back to the project view, and then click on the next diagram thumbnail.

Again this is good feedback, and I will put it into the FAQs.

Cheers, Jon

**Comment by jgadsden** [comment]
That is a very good point - it is misleading. I have changed it to 'Close Diagram' in commit:

https://github.com/OWASP/threat-dragon-core/commit/51263f85d98b26dee7f445ccc23ca64934fa1a74

which should appear in version 1.3 due in August

**Comment by craig-Ayla** [comment]
Thank you!

Craig




On Jul 16, 2020, at 1:54 AM, Jon Gadsden <notifications@github.com<mailto:notifications@github.com>> wrote:



That is a very good point - it is misleading. I have changed it to 'Close Diagram' in commit:

OWASP/threat-dragon-core@51263f8<https://github.com/OWASP/threat-dragon-core/commit/51263f85d98b26dee7f445ccc23ca64934fa1a74>

which should appear in version 1.3 due in August
